src/GraphQL.Benchmarks/Benchmarks/DetailedBenchmark.cs

```diff
@@ -0,0 +1,269 @@
+                ExecutionResult = Execute();
+                _ = Serialize();
+            }
+
+            private static readonly GraphQLParser.Parser _parser = new GraphQLParser.Parser(new GraphQLParser.Lexer());
+
+            public ISchema BuildSchema()
+            {
+                return SchemaBuilder();
+            }
+
+            public GraphQLDocument Parse()
+            {
+                return _parser.Parse(new GraphQLParser.Source(Query));
+            }
+
+            public Document Convert()
+            {
+                return CoreToVanillaConverter.Convert(GraphQLDocument);
+            }
+
+            public Language.AST.Variables ParseVariables()
+            {
+                return Inputs == null ? null : ExecutionHelper.GetVariableValues(Document, Schema, Operation.Variables, Inputs);
+            }
+
+            private static readonly DocumentValidator _documentValidator = new DocumentValidator();
+            public IValidationResult Validate()
+            {
+                return _documentValidator.ValidateAsync(
+                    Query,
+                    Schema,
+                    Document,
+                    null,
+                    null,
+                    Inputs).Result;
+            }
+
+            private static readonly ParallelExecutionStrategy _parallelExecutionStrategy = new ParallelExecutionStrategy();
+            public ExecutionResult Execute()
+            {
+                var context = new ExecutionContext
+                {
+                    Document = Document,
+                    Schema = Schema,
+                    RootValue = null,
+                    UserContext = new Dictionary<string, object>(),
+
+                    Operation = Operation,
+                    Variables = Variables,
+                    Fragments = Document.Fragments,
+                    Errors = new ExecutionErrors(),
+                    Extensions = new Dictionary<string, object>(),
+                    CancellationToken = default,
+
+                    Metrics = new Instrumentation.Metrics(false),
+                    Listeners = new List<IDocumentExecutionListener>(),
+                    ThrowOnUnhandledException = true,
+                    UnhandledExceptionDelegate = context => { },
+                    MaxParallelExecutionCount = int.MaxValue,
+                    RequestServices = null
+                };
+                return _parallelExecutionStrategy.ExecuteAsync(context).Result;
+            }
+
+            private static readonly DocumentWriter _documentWriter = new DocumentWriter();
+            public System.IO.MemoryStream Serialize()
+            {
+                var mem = new System.IO.MemoryStream();
+                _documentWriter.WriteAsync(mem, ExecutionResult).GetAwaiter().GetResult();
+                mem.Position = 0;
+                return mem;
+            }
+        }
+        private class BenchmarkConfig : ManualConfig
```

Comment:
Please move it on top.

Reply:
On top of what?  Top of the file?  Out of the class?

Reply:
I did so.  But why?!?  This is superfluous code that nobody wants to read.  It's too bad I had to spend more than a half hour just trying to get it to return the results in the correct order.  Now if anyone wants to add a bunchmark, they have to scroll past all this garbage to get to the spot where they can add the additional benchmark.

Reply:
This is why I would not immediately use the command line parser, by the way.  It might take me another half hour to get it to work the way I want it to, when I was able to write 30 lines of code that works exactly the way I want it to the first time.  It's not worth the time, especially when there are so little benefits.

Reply:
I could spend that time on many of the outstanding issues or PRs, rather than messing with a third-party library I'm unfamiliar with.

Reply:
>  I had to spend more than a half hour just trying to get it to return the results in the correct order

Me too.